modules/openedx/s3/course_discovery.tf

```
@@ -0,0 +1,52 @@
+data aws_iam_policy_document edxapp_course_discovery_s3_full_access {
+  count = var.enable_course_discovery ? 1 : 0
+  version = "2012-10-17"
+  statement {
+    actions = [
+      "s3:*",
+    ]
+    effect = "Allow"
+
+    resources = [
+      aws_s3_bucket.edxapp_course_discovery.arn,
+      "${aws_s3_bucket.edxapp_course_discovery.arn}/*",
```

When using count, you'll need to add an index to the element you're fetching:
```suggestion
      aws_s3_bucket.edxapp_course_discovery[0].arn,
      "${aws_s3_bucket.edxapp_course_discovery[0].arn}/*",
```

---

Makes sense